src/nutrients_plankton_bacteria_detritus.jl

```diff
@@ -85,70 +97,106 @@ end
 #   -> P⁺ = P⁻ / (1 + Δt * m)
 #
 
-@inline bacteria_production(μᵇ, kᴰ, D, B) = μᵇ * D / (D + kᴰ) * B
-@inline plankton_production(μᵖ, kᴺ, kᴵ, I, N, P) = μᵖ * (N / (N + kᴺ) * I / (I + kᴵ)) * P
-@inline bacteria_mortality(m, B) = m * B^2
-@inline plankton_mortality(m, P) = m * P^2
+# A depth-dependent temperature curve from Zakem (2018)
+# Temp = 12 .*exp.(z./ 150) .+ 12 .*exp.(z ./ 500) .+ 2
+
+# Temperature modification to metabolic rates, following the Arrhenius equation
+# @inline temp_fun(Temp) = 0.8 .* exp.(-4000 .*(1 ./ (Temp .+ 273.15) .- 1 ./ 293.15))
+
+@inline bacteria_production(μᵇ, kᴰ, D, B) = (μᵇ * D / (D + kᴰ) * B) 
+@inline phytoplankton_production(μᵖ, kᴺ, kᴵ, I, N, P) = (μᵖ * min(N / (N + kᴺ) , I / (I + kᴵ)) * P) 
+@inline zooplankton_production(gₘ, kᴾ, γ, P, Z) = γ * (gₘ * P / (P + kᴾ) * Z) 
+@inline bacteria_mortality(mlin,mq, B) = mlin * B + mq * B^2
+@inline phytoplankton_mortality(mlin,mq, P) = mlin * P + mq * P^2
+@inline zooplankton_mortality(mlin,mq, Z) = mlin * Z + mq * Z^2
 
 @inline function (bgc::NutrientsPlanktonBacteriaDetritus)(i, j, k, grid, ::Val{:N}, clock, fields)
     μᵖ = bgc.maximum_plankton_growth_rate
     μᵇ = bgc.maximum_bacteria_growth_rate
+    gₘ = bgc.maximum_grazing_rate
     kᴰ = bgc.detritus_half_saturation
     kᴺ = bgc.nutrient_half_saturation
+    kᴾ = bgc.phytoplankton_half_saturation
     kᴵ = bgc.PAR_half_saturation
     λ = bgc.PAR_attenuation_scale
-    m = bgc.quadratic_mortality_rate
     y = bgc.bacteria_yield
+    γ = bgc.zooplankton_assimilation
 
     # Available photosynthetic radiation
     z = znode(i, j, k, grid, c, c, c)
-    I = exp(z / λ)
+
+    # TODO: design a user interface for prescribing incoming shortwave
+    I = 700 * exp(z / λ)
 
     P = @inbounds fields.P[i, j, k]
+    Z = @inbounds fields.Z[i, j, k]
     D = @inbounds fields.D[i, j, k]
     B = @inbounds fields.B[i, j, k]
     N = @inbounds fields.N[i, j, k]
     
-    return - plankton_production(μᵖ, kᴺ, kᴵ, I, N, P) + bacteria_production(μᵇ, kᴰ, D, B) * (1/y - 1)
+    return - phytoplankton_production(μᵖ, kᴺ, kᴵ, I, N, P) + bacteria_production(μᵇ, kᴰ, D, B) * (1/y - 1) + zooplankton_production(gₘ, kᴾ, γ, P, Z) * (1/γ - 1)
 end
 
 @inline function (bgc::NutrientsPlanktonBacteriaDetritus)(i, j, k, grid, ::Val{:P}, clock, fields)
     μᵖ = bgc.maximum_plankton_growth_rate
+    gₘ = bgc.maximum_grazing_rate
     kᴺ = bgc.nutrient_half_saturation
     kᴵ = bgc.PAR_half_saturation
+    kᴾ = bgc.phytoplankton_half_saturation
     λ = bgc.PAR_attenuation_scale
-    m = bgc.quadratic_mortality_rate
+    mlin = bgc.linear_mortality_rate
+    mq = bgc.quadratic_mortality_rate
```

Comment:
Nice to clarify the quadratic rate! I think we will need a separate one for the zooplankton, since for that type we are not explicitly representing their grazer. Let's also talk about whether to/how to include grazing on bacteria.